### PR TITLE
Same available_versions, parameters, .. for lisflood and marrmot

### LINKED ISSUE #61: Conflict about the naming of work_dir and cfg_dir
There is a conflict about the naming of `work_dir` (which is what grpc4bmi uses) and `cfg_dir` (which is what [PyMT uses](https://pymt.readthedocs.io/en/latest/usage.html)). In PyMT usage, the variable is named as `cfg_dir`. However, it is referred to `run` directory in the text. (cc @Peter9192)

### LINKED ISSUE #112: Return ewatercycle.models.available_versions in a tuple
We want `ewatercycle.models.available_versions` as a `tuple` instead of a `list`. This is not currently the case for [lisflood](https://github.com/eWaterCycle/ewatercycle/blob/955499d89f30a0d4a45dd26037ee9caae1a6206d/ewatercycle/models/lisflood.py#L59), [marrmotm01](https://github.com/eWaterCycle/ewatercycle/blob/955499d89f30a0d4a45dd26037ee9caae1a6206d/ewatercycle/models/marrmot.py#L62), [marrmotm14](https://github.com/eWaterCycle/ewatercycle/blob/955499d89f30a0d4a45dd26037ee9caae1a6206d/ewatercycle/models/marrmot.py#L290) and [api_notebook](https://github.com/eWaterCycle/ewatercycle/pull/81).

### LINKED ISSUE #113: Remove path from model.parameters
We dont want path in `model.parameters` like forcing directory. It should be fixed in [lisflood](https://github.com/eWaterCycle/ewatercycle/blob/955499d89f30a0d4a45dd26037ee9caae1a6206d/ewatercycle/models/lisflood.py#L245), [marrmotm01](https://github.com/eWaterCycle/ewatercycle/blob/955499d89f30a0d4a45dd26037ee9caae1a6206d/ewatercycle/models/marrmot.py#L251), and [marrmotm14](https://github.com/eWaterCycle/ewatercycle/blob/955499d89f30a0d4a45dd26037ee9caae1a6206d/ewatercycle/models/marrmot.py#L504).

### LINKED ISSUE #114: Marrmot class: Change warnings.warn to a print statement
Currently [marrmot class](https://github.com/eWaterCycle/ewatercycle/blob/955499d89f30a0d4a45dd26037ee9caae1a6206d/ewatercycle/models/marrmot.py#L407) uses `warnings.warn` about the parameters inside the forcing file. It can be a print message.

### PR DESCRIPTION
**Pull request by SarahAlidoost**
closes #112, closes #113, closes #114, closes #61

### CODE REVIEW

**Comment by SarahAlidoost** [comment]
@Peter9192 and @sverhoeven who has time to review this PR? thanks.

File: ewatercycle/models/lisflood.py [hunk 1]
```diff
@@ -299,20 +296,20 @@ def parameters(self) -> Iterable[Tuple[str, Any]]:
 #     return output_dir
 
 
-def _generate_workdir(work_dir: Path = None) -> Path:
+def _generate_workdir(cfg_dir: Path = None) -> Path:
     """
 
     Args:
-        work_dir: If work dir is None then create sub-directory in CFG['output_dir']
+        cfg_dir: If work dir is None then create sub-directory in CFG['output_dir']
```
Comment (sverhoeven) [inline ewatercycle/models/lisflood.py]:
```suggestion
        cfg_dir: If cfg dir is None then create sub-directory in CFG['output_dir']
```

File: ewatercycle/models/marrmot.py [hunk 2]
```diff
@@ -27,18 +28,18 @@ class Solver:
     resnorm_maxiter: float = 6.0
 
 
-def _generate_work_dir(work_dir: Path = None) -> Path:
+def _generate_cfg_dir(cfg_dir: Path = None) -> Path:
     """
     Args:
-        work_dir: If work dir is None or does not exist then create sub-directory in CFG['output_dir']
+        cfg_dir: If work dir is None or does not exist then create sub-directory in CFG['output_dir']
```
Comment (sverhoeven) [inline ewatercycle/models/marrmot.py]:
```suggestion
        cfg_dir: If cfg dir is None or does not exist then create sub-directory in CFG['output_dir']
```

**Review by sverhoeven — APPROVED** [review]
Looks like all of the issues have indeed been solved by this PR. great work.

Just 2 small suggestion in docstring of private methods.

**Pull request merged**